projects/com.oracle.truffle.llvm.parser/src/com/oracle/truffle/llvm/parser/model/enums/BinaryOperator.java

```diff
@@ -61,12 +61,23 @@ public static BinaryOperator decode(int opcode, boolean isFloatingPoint) {
     }
 
     private final int fpmap;
+    private final String name;
 
-    BinaryOperator(int fpmap) {
+    BinaryOperator(int fpmap, String name) {
         this.fpmap = fpmap;
+        this.name = name;
     }
 
     private BinaryOperator fp() {
         return fpmap < 0 ? null : values()[fpmap];
     }
+
+    public boolean isFloatingPoint() {
+        return this.ordinal() > INT_XOR.ordinal();
```

Comment:
this method is needed to differ between normal and floating point operators in LLVMIR writer.

If this is not sensible to put it into the enum, I can make helper classes for such tasks in my library and this PR would be no longer required at all.